sdk/eventhubs/azure_messaging_eventhubs/examples/eventhubs_tracing_subscriber.rs

```diff
@@ -0,0 +1,217 @@
+            .add_property(IS_EVENT_PROPERTY.to_string(), metadata.is_event())
+            .add_property(IS_SPAN_PROPERTY.to_string(), metadata.is_span());
+
+        // Handle optional metadata fields.
+        if let Some(module_path) = metadata.module_path() {
+            event_data_builder =
+                event_data_builder.add_property(MODULE_PATH_PROPERTY.to_string(), module_path);
+        }
+        if let Some(file) = metadata.file() {
+            event_data_builder = event_data_builder.add_property(FILE_PROPERTY.to_string(), file);
+        }
+        if let Some(line) = metadata.line() {
+            event_data_builder = event_data_builder.add_property(LINE_PROPERTY.to_string(), line);
+        }
+
+        let event_data = event_data_builder.build();
+
+        // Send the event to Event Hubs asynchronously
+        self.sender.try_send(event_data).unwrap_or_else(|_| {
+            eprintln!("Failed to send event to EventHubs");
+        });
+    }
+}
+
+/// A visitor that formats log messages and collects them into a buffer.
+struct EventVisitor {
+    buffer: String,
+}
+
+impl Visit for EventVisitor {
+    fn record_str(&mut self, field: &tracing::field::Field, value: &str) {
+        println!("record_str: {}: {}", field.name(), value);
+        // Append the field name and value to the buffer
+        self.buffer
+            .push_str(&format!("{}: {}", field.name(), value));
+    }
+
+    fn record_debug(&mut self, field: &tracing::field::Field, value: &dyn std::fmt::Debug) {
+        println!("record_debug: {}: {:?}", field.name(), value);
+        self.buffer
+            .push_str(&format!("{}: {:?}", field.name(), value));
+    }
+}
+
+/// Test struct to demonstrate structured data logging.
+#[derive(Debug)]
+struct StructuredData {
+    body: String,
+    properties: std::collections::HashMap<String, String>,
+}
+
+impl std::fmt::Display for StructuredData {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "StructuredData: {}", self.body)
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    // Get EventHubs fully qualified domain name and hub name from environment variables
+    let fully_qualified_domain_name =
+        env::var("EVENTHUBS_HOST").expect("EVENTHUBS_HOST must be set");
+    let eventhub_name = env::var("EVENTHUB_NAME").expect("EVENTHUB_NAME must be set");
+
+    // Create our custom EventHubsLayer
+    let eventhubs_layer = EventHubsLayer::new(&fully_qualified_domain_name, &eventhub_name).await?;
+
+    // Set up tracing subscriber with both console and EventHubs outputs. Note that the components of the EventHubs service (which also uses tracing)
+    // need to be disabled.
+    let subscriber = Registry::default()
+        .with(fmt::layer().with_target(true))
+        .with(eventhubs_layer)
+        .with(
+            tracing_subscriber::EnvFilter::from_default_env()
+                .add_directive(Level::DEBUG.into())
```

Comment:
Would this override the default from `from_default_env()`?

Reply:
Yes.